### PR TITLE
feat(openid): add /.well-known/openid-configuration route

### LINKED ISSUE #3299: Add openid-connect-compatible discovery document
The OpenID Connect spec defines a discovery protocol through which reliers can learn about the various endpoints and supported features of the IdP [1].  We should implement such a thing for Firefox Accounts.  I think it's most sensible deployed as part of the content-server API because this allows reliers to discover all our other endpoint URLs just by knowing "accounts.firefox.com".

[1] https://github.com/mozilla/fxa/tree/master/features/FxA-38-openid-connect#discovery

### PR DESCRIPTION
**Pull request by seanmonstar**
Closes #3299

### CODE REVIEW

File: server/lib/configuration.js [hunk 1]
```diff
@@ -370,6 +370,20 @@ var conf = module.exports = convict({
     env: 'FXA_OAUTH_URL',
     format: 'url'
   },
+  openid_configuration: {
+    response_types_supported: ['code', 'token'],
```
Comment (shane-tomlinson) [inline server/lib/configuration.js]: Can you alphabetize the keys in the `openid_configuration` object?

File: server/lib/routes/get-openid-configuration.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+var config = require('../configuration');
+
+var issuer = config.get('public_url');
+var authorizationEndpoint = config.get('oauth_url') + '/v1/authorization';
+var tokenEndpoint = config.get('oauth_url') + '/v1/token';
+var userInfoEndpoint = config.get('profile_url') + '/v1/profile';
+var jwksEndpoint = config.get('oauth_url') + '/v1/jwks';
+
+var openidConfig = {
+  issuer: issuer,
```
Comment (shane-tomlinson) [inline server/lib/routes/get-openid-configuration.js]: Alphabetize the keys here too.

File: server/lib/routes/get-openid-configuration.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+
+var config = require('../configuration');
+
+var issuer = config.get('public_url');
```
Comment (shane-tomlinson) [inline server/lib/routes/get-openid-configuration.js]: Unless one variable depends on another, might as well alphabetize the variable names too.

File: server/lib/routes/get-openid-configuration.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+module.exports = function (config) {
+  var route = {};
+  route.method = 'get';
+  route.path = '/.well-known/openid-configuration';
+
+  route.process = function (req, res) {
+
+    // taken from https://accounts.google.com/.well-known/openid-configuration
+    res.header('Cache-Control', 'public, max-age=3600');
```
Comment (shane-tomlinson) [inline server/lib/routes/get-openid-configuration.js]: Should the max-age be made into a configuration variable?

File: server/lib/configuration.js (hunk 1 above)
Comment (seanmonstar) [inline server/lib/configuration.js]: Yea, I see the lint complaining. I initially wrote them in the same order as was in Google's doc. Easy to rearrange.

**Comment by vladikoff** [comment]
@seanmonstar do we need this in train-52?

**Comment by vladikoff** [comment]
Closing in favour of https://github.com/mozilla/fxa-content-server/pull/3322

**Pull request closed without merging**